### PR TITLE
#1036 Interact Caching Choices

### PR DESCRIPTION
**Pull request by mmilutinovic1313**
Caching is now active for Interact. Added update hooks for each of the files that were changed. Issue: `CIS User List` does not seem to exist in our CIS folder structure. Details of the caching times are listed below:

Interact Choices:

- [x] `CIS Authority Courses` = 1 day
- [x] `CIS Section list` = 1 day
- [x] `CIS User list` = 1 day (`CIS User List` folder or exported views feature file does not currently exist...will make issue to either create or find where it went.)
- [x] `icor object lists` = 1 day
- [x] `ICOR Player EVA` = 30 min
- [x] `Timeline` = 30 min
- [x] `Timeline Content type` = 30 min

#1055

### CODE REVIEW

File: core/dslmcode/shared/drupal-7.x/modules/elmsln_contrib/cis_connector/modules/authorities/cis_course_authority/cis_course_authority.install [hunk 1]
```diff
@@ -22,4 +22,16 @@ function cis_course_authority_update_7001(&$sandbox) {
 function cis_course_authority_update_7002(&$sandbox) {
   module_enable(array('ds'));
   features_revert_module('cis_course_authority');
+}
+/**
```
Comment (btopro) [inline core/dslmcode/shared/drupal-7.x/modules/elmsln_contrib/cis_connector/modules/authorities/cis_course_authority/cis_course_authority.install]: double comment here

File: core/dslmcode/shared/drupal-7.x/modules/elmsln_contrib/cis_connector/modules/features/cis_section/cis_section.views_default.inc [hunk 2]
```diff
@@ -190,107 +194,6 @@ function cis_section_views_default_views() {
   $handler->display->display_options['menu']['name'] = 'menu-elmsln-settings';
   $handler->display->display_options['menu']['context'] = 0;
   $handler->display->display_options['menu']['context_only_inline'] = 0;
-  $export['cis_section_list'] = $view;
```
Comment (btopro) [inline core/dslmcode/shared/drupal-7.x/modules/elmsln_contrib/cis_connector/modules/features/cis_section/cis_section.views_default.inc]: Did you drop `cis/user-list`, I'm confused now about it wasn't in place previously?

File: core/dslmcode/stacks/interact/sites/all/modules/local_contrib/views_timelinejs/views_timelinejs.install [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?php
+/**
+ * @file
+ * Code for the Views TimelineJS.
+ */
+function views_timelinejs_update_7001(&$sandbox) {
+/**
+ * Implements hook_update_N().
```
Comment (btopro) [inline core/dslmcode/stacks/interact/sites/all/modules/local_contrib/views_timelinejs/views_timelinejs.install]: double comment again

File: core/dslmcode/stacks/interact/sites/all/modules/local_contrib/views_timelinejs_extras/views_timelinejs_extras.install [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?php
+/**
+ * @file
+ * Code for the Views TimelineJS Extras.
+ */
+function views_timelinejs_extras_update_7001(&$sandbox) {
+/**
+ * Implements hook_update_N().
```
Comment (btopro) [inline core/dslmcode/stacks/interact/sites/all/modules/local_contrib/views_timelinejs_extras/views_timelinejs_extras.install]: double comment

File: core/dslmcode/shared/drupal-7.x/modules/elmsln_contrib/cis_connector/modules/features/cis_section/cis_section.views_default.inc (hunk 2 above)
Comment (mmilutinovic1313) [inline core/dslmcode/shared/drupal-7.x/modules/elmsln_contrib/cis_connector/modules/features/cis_section/cis_section.views_default.inc]: Good catch - cis/user-list was never there. I discovered this when I did the export of user-list view and I couldn't find the right file to put it's feature in. I accidently copied some of it's code in here so I'll have to go revert it to what it was and re-export.

**Comment by mmilutinovic1313** [comment]
Okay so I found something interesting in the code for the `cis_section.views_default.inc` file that you commented on. I looked through the history of the file. That file creates a property at the beginning with a name of `'cis_section_list'`

However, the property that was being returned at the end of the file was never the `'cis_section_list'` property. Instead it was `'cis_user_list'`. That brings me to 2 questions:

1) Which property is the correct one for the line 294? Is it:
` $export['cis_user_list'] = $view;` or   `$export['cis_section_list'] = $view;` (edited)
2) What do we do with the export that handles `cis_user_list` as the code in the `cis_section.views_default.inc` file is from the `cis_section_list` view. Currently no files exist that contain export code from `cis_user_list` view.

**Comment by mmilutinovic1313** [comment]
Okay this is good to go. I had a brain malfunction and never should have deleted the user_list. This PR is good to go now

**Pull request merged**